tuxemon/formula.py

```diff
@@ -237,3 +239,40 @@ def battle_math(player: NPC, output: str) -> None:
         player["percent_draw"] = round(
             (player["battle_draw"] / player["battle_total"]) * 100
         )
+
+
+def rematch(
+    player: NPC,
+    opponent: NPC,
+    monster: Monster,
+    output: OutputBattle,
+    date: int,
+) -> None:
+    today = dt.date.today().toordinal()
+    diff_date = today - date
+    low = player.game_variables["party_level_highest"]
+    high = 0
+    # nr days between match and rematch
+    if diff_date == 0:
+        high = low + 1
+    elif diff_date < 10:
+        high = low + 2
+    elif diff_date < 50:
+        high = low + 3
+    else:
+        high = low + 5
+    # different conditions related to the output
+    if output.won:
+        monster.level = random.randint(low, high)
```

Comment:
My point was that you have the exact same behaviour on lines 266, 268 and 270, so what's the point of the if/else?

Reply:
I thought about differentiate for output, but at the end is going to happen only after the output.won, so yeah, I'm going to remove it.